sklearn/feature_selection/mutual_info.py

```diff
@@ -0,0 +1,230 @@
+# Authors: Andrea Bravi <a.bravi@uottawa.ca>
+#          Nikolay Mayorov <n59_ru@hotmail.com>
+# License: 3-clause BSD
+
+from __future__ import division
+
+import numpy as np
+from scipy.sparse import issparse
+from scipy.special import digamma
+
+from ..base import BaseEstimator
+from .base import SelectorMixin
+from ..metrics.cluster.supervised import mutual_info_score
+from ..utils.validation import check_X_y
+from ..neighbors import NearestNeighbors
+
+
+def _compute_mi_cc(x, y, n_neighbors):
+    n_samples = x.size
+
+    x_std = np.std(x)
+    if x_std > 0:
+        x = x / x_std
+    x = x.reshape((-1, 1))
+
+    y_std = np.std(y)
+    if y_std > 0:
+        y = y / y_std
+    y = y.reshape((-1, 1))
+
+    xy = np.hstack((x, y))
+
+    nn = NearestNeighbors(metric='chebyshev', n_neighbors=n_neighbors)
+
+    nn.fit(xy)
+    radius = nn.kneighbors()[0]
+    radius = np.nextafter(radius[:, -1], 0)
+
+    nn.set_params(algorithm='kd_tree')
+
+    nn.fit(x)
+    ind = nn.radius_neighbors(radius=radius, return_distance=False)
+    nx = np.array([i.size for i in ind])
+
+    nn.fit(y)
+    ind = nn.radius_neighbors(radius=radius, return_distance=False)
+    ny = np.array([i.size for i in ind])
+
+    mi = (digamma(n_samples) + digamma(n_neighbors) -
+          np.mean(digamma(nx + 1)) - np.mean(digamma(ny + 1)))
+
+    return max(0, mi)
+
+
+def _compute_mi_cd(c, d, n_neighbors):
+    c = c.reshape((-1, 1))
+    n_samples = c.size
+
+    nn = NearestNeighbors(n_neighbors=n_neighbors)
+
+    radius = np.empty(n_samples)
+    label_counts = np.empty(n_samples)
+    for label in np.unique(d):
+        mask = d == label
+        count = np.sum(mask)
+        if count > 1:
+            nn.set_params(n_neighbors=min(n_neighbors, count - 1))
+            nn.fit(c[mask])
+            r = nn.kneighbors()[0]
+            radius[mask] = np.nextafter(r[:, -1], 0)
+        else:
+            radius[mask] = 0
+        label_counts[mask] = count
+
+    nn.set_params(algorithm='kd_tree')
+    nn.fit(c)
+    ind = nn.radius_neighbors(radius=radius, return_distance=False)
+    neighbor_counts = np.array([i.size for i in ind])
+
+    mi = (digamma(n_samples) + digamma(n_neighbors) -
+          np.mean(digamma(label_counts)) -
+          np.mean(digamma(neighbor_counts + 1)))
+
+    return max(0, mi)
+
+
+def _compute_mi(x, y, x_categorical, y_categorical, n_neighbors=3):
```

Comment:
Would be great if you could add documentation for this and the other helper functions above, (though they are private)